tests/datasets/test__init__.py

```diff
@@ -0,0 +1,424 @@
+                "project-path",
+                project_id=project_id,
+                show_hidden=True,
+                object_client=object_client,
+            ),
+        ]
+    )
+    _get_file_mock.assert_called_once_with(
+        "project-path/test-file", local_dests[1], project_id, object_client
+    )
+
+
+def test_datasets_put_file_and_create_parent_directories(mocker, mock_client):
+    object_client, project_id = mock_client
+
+    path_project_parent_directory = mocker.patch(
+        "faculty.datasets.path.project_parent_directory", return_value="/"
+    )
+    object_client.create_directory.return_value = mocker.Mock()
+
+    os_path_isdir_mock = mocker.patch("os.path.isdir", return_value=False)
+
+    upload_result_mock = mocker.Mock()
+    upload_mock = mocker.patch(
+        "faculty.datasets.transfer.upload_file",
+        return_value=upload_result_mock,
+    )
+
+    datasets.put("local-path", "project-path", project_id)
+
+    path_project_parent_directory.assert_called_once_with("project-path")
+    object_client.create_directory.assert_called_once_with(
+        project_id, "/", parents=True
+    )
+    os_path_isdir_mock.assert_called_once_with("local-path")
+    upload_mock.assert_called_once_with(
+        object_client, project_id, "project-path", "local-path"
+    )
+
+
+def test_datasets_put_directory(mocker, mock_client):
+    object_client, project_id = mock_client
+
+    path_project_parent_directory = mocker.patch(
+        "faculty.datasets.path.project_parent_directory", return_value="/"
+    )
+    object_client.create_directory.return_value = mocker.Mock()
+
+    os_path_isdir_mock = mocker.patch(
+        "os.path.isdir", side_effect=[True, False]
+    )
+
+    entry_mock = "test-file"
+    os_listdir_mock = mocker.patch("os.listdir", return_value=[entry_mock])
+
+    _put_file_mock = mocker.patch(
+        "faculty.datasets._put_file", return_value=mocker.Mock()
+    )
+
+    datasets.put("local-path", "project-path", project_id)
+
+    path_project_parent_directory.assert_called_once_with("project-path")
+    object_client.create_directory.assert_has_calls(
+        [
+            mocker.call(project_id, "/", parents=True),
+            mocker.call(project_id, "project-path"),
+        ]
+    )
+    os_path_isdir_mock.assert_has_calls(
+        [mocker.call("local-path"), mocker.call("local-path/test-file")]
+    )
+    os_listdir_mock.assert_called_once_with("local-path")
+    _put_file_mock.assert_called_once_with(
+        "local-path/test-file",
+        "project-path/test-file",
+        project_id,
+        object_client,
+    )
+
+
+def test_datasets_cp_path(mocker, mock_client):
+    object_client, project_id = mock_client
+    object_client.copy.return_value = mocker.Mock()
+
+    path_project_parent_directory = mocker.patch(
+        "faculty.datasets.path.project_parent_directory", return_value="/"
+    )
+    object_client.create_directory.return_value = mocker.Mock()
```

Comment:
This is essentially a noop. The return value will be a `Mock` object by default.